### ISSUE TITLE
How to fill between two lines in y direction only within intersection part?

### ISSUE THREAD

**Issue by ilongshan** [plotly/plotly.py#4311]
This code snippet will plot two lines:

```python
import plotly.graph_objs as go

anchor_rate = [1000, 1660.89, 3220.62, 5927.36]
anchor_psnr = [31.799098, 34.155477, 36.053423, 36.5]

tested_rate = [752.33,1608.90,3114.10,5685.72]
tested_psnr = [31.336015,33.752160,35.730303,37.200365]

fig = go.Figure()

fig.add_trace(go.Scatter(
    x=anchor_rate,
    y=anchor_psnr,
))

fig.add_trace(go.Scatter(
    x=tested_rate,
    y=tested_psnr,
))

fig.show()
```

This is the plot:
![newplot (1)](https://github.com/plotly/plotly.py/assets/7447401/e2005ea9-2d77-42a3-a7b6-499a4ed30d0a)

How to fill between two lines in y direction only within intersection part, like this? And how can fill them with different colors?
![newplot](https://github.com/plotly/plotly.py/assets/7447401/22d732ed-770c-4eaf-ae47-199edb133711)

In summary, two questions:
1. How to fill between two lines in y direction only within intersection part?
2. And how can fill them with different colors?

**Comment by sOnU1002** [comment]
import plotly.graph_objs as go

anchor_rate = [1000, 1660.89, 3220.62, 5927.36]
anchor_psnr = [31.799098, 34.155477, 36.053423, 36.5]

tested_rate = [752.33, 1608.90, 3114.10, 5685.72]
tested_psnr = [31.336015, 33.752160, 35.730303, 37.200365]

fig = go.Figure()

fig.add_trace(go.Scatter(
    x=anchor_rate,
    y=anchor_psnr,
    name='Anchor',
))


fig.add_trace(go.Scatter(
    x=tested_rate,
    y=tested_psnr,
    name='Tested',
))

fig.add_trace(go.Scatter(
    x=anchor_rate + tested_rate[::-1],
    y=anchor_psnr + tested_psnr[::-1],
    fill='tozerox',  # Fill to the x-axis
    fillcolor='rgba(0, 100, 80, 0.2)',  # Specify fill color
    line=dict(color='rgba(255,255,255,0)'),  # Set line color to transparent
    showlegend=False,  # Hide from legend
))

fig.show()

**Comment by ilongshan** [comment]
@sOnU1002 Thanks very much for your reply. But it seems not the expected plots:
1. It should only fill the intersection part of the two lines, not beyond them on the left.
2. It should fill with the horizontal line, not linking the two end points on the right.
Still thanks for your solution.
<img width="1316" alt="image" src="https://github.com/plotly/plotly.py/assets/7447401/f493508c-650a-4d2c-bfdb-4f40e09aabef">

**Comment by sOnU1002** [comment]
ook will try also if any update i will let u know

**Comment by ilongshan** [comment]
You can get the intersection points of the horizontal line with the curve with linear Interpolation.
<img width="1286" alt="image" src="https://github.com/plotly/plotly.py/assets/7447401/fbc07627-e6c6-4e25-85fb-39fbf8208baa">

**Comment by sOnU1002** [comment]
Thanks, I'll check it out.

**Comment by gvwilson** [comment]
Hi - we are tidying up stale issues and PRs in Plotly's public repositories so that we can focus on things that are still important to our community. Since this one has been sitting for a while, I'm going to close it; if it is still a concern, please add a comment letting us know what recent version of our software you've checked it with so that I can reopen it and add it to our backlog. If you'd like to submit a PR, we'd be happy to prioritize a review, and if it's a request for tech support, please post in our community forum. Thank you - @gvwilson

**Comment by sOnU1002** [comment]
Please sir don't close it

**Comment by sOnU1002** [comment]
Give me some if update u regarding it

**Comment by gvwilson** [comment]
Thanks @sOnU1002 - I'll reopen this and add it to our backlog, but realistically, nobody is going to be able to address it any time soon.

**Comment by sOnU1002** [comment]
Sure

**Comment by sOnU1002** [comment]
import numpy as np
import plotly.graph_objs as go

# Define the data
anchor_rate = np.array([1000, 1660.89, 3220.62, 5927.36])
anchor_psnr = np.array([31.799098, 34.155477, 36.053423, 36.5])

tested_rate = np.array([752.33, 1608.90, 3114.10, 5685.72])
tested_psnr = np.array([31.336015, 33.752160, 35.730303, 37.200365])

# Find intersection points
intersection_x = np.linspace(max(min(anchor_rate), min(tested_rate)), min(max(anchor_rate), max(tested_rate)), 500)
intersection_y1 = np.interp(intersection_x, anchor_rate, anchor_psnr)
intersection_y2 = np.interp(intersection_x, tested_rate, tested_psnr)

# Create the plot
fig = go.Figure()

# Add the anchor and tested lines
fig.add_trace(go.Scatter(
    x=anchor_rate,
    y=anchor_psnr,
    mode='lines',
    name='Anchor'
))

fig.add_trace(go.Scatter(
    x=tested_rate,
    y=tested_psnr,
    mode='lines',
    name='Tested'
))

# Add filled area between the lines
fig.add_trace(go.Scatter(
    x=np.concatenate([intersection_x, intersection_x[::-1]]),
    y=np.concatenate([intersection_y1, intersection_y2[::-1]]),
    fill='toself',
    fillcolor='rgba(0, 100, 80, 0.2)',
    line=dict(color='rgba(255,255,255,0)'),
    name='Intersection Area'
))

# Update layout
fig.update_layout(
    title='Plot with Filled Intersection Area',
    xaxis_title='Rate',
    yaxis_title='PSNR'
)

fig.show()
please check it once sir
